extensions/amp-video-docking/0.1/timeout.js

```diff
@@ -16,15 +16,33 @@
 
 import {Services} from '../../../src/services';
 
+/**
+ * @typedef {{
+ *   setTimeout: function(function(), time):number,
+ *   clearTimeout: function(number),
+ * }}
+ */
+let TimerDef;
+
+/**
+ * @param {!Window} win
+ * @return {!TimerDef}
+ */
+const usingTimerService = (win) => ({
+  setTimeout: (handler, time) => Services.timerFor(win).delay(handler, time),
+  clearTimeout: (id) => Services.timerFor(win).cancel(id),
+});
```

Comment:
Reusing 0.1 utility here, with these dependencies injected to that the Preact version uses plain `setTimeout`.